algosdk/abi/interface.py

```diff
@@ -0,0 +1,39 @@
+import json
+
+from algosdk.abi.method import Method
+
+
+class Interface:
+    """
+    Represents a ABI interface description.
+
+    Args:
+        name (string): name of the interface
+        methods (list): list of Method objects
+    """
+
+    def __init__(self, name, methods) -> None:
```

Comment:
Continuing with my first comment, If we are gradually transitioning to using type hints, this new class presents an opportunity for that. If not, I'm not sure the purpose of having partial type hints. `__eq__` is a lonely outlier.

Reply:
Agree with your point, I will clean this up